Added new Unit tests for methods in CertifyManager.cs

jrnelson90 opened the pull request.

jrnelson90 committed:
Added new Unit tests for methods in CertifyManager.cs

webprofusion-chrisc commented on src/Certify.Tests/Certify.Core.Tests.Unit/CertifyManagerTests.cs:
```diff
@@ -0,0 +1,174 @@
+
+            // Validate return from CertifyManager.GetACMEProvider()
+            Assert.IsNotNull(acmeClientProvider, "Expected response from CertifyManager.GetACMEProvider() to not be null");
+            Assert.AreEqual("https://acme-staging-v02.api.letsencrypt.org/directory", acmeClientProvider.GetAcmeBaseURI());
```
This is assuming that we always have an LE staging account as our default (and it's created), we should probably define the expected CA directory URL in test config somewhere.

jrnelson90 replied:
This has been refactored to grab value from CertificateAuthority.CoreCertificateAuthorities

webprofusion-chrisc commented on src/Certify.Tests/Certify.Core.Tests.Unit/CertifyManagerTests.cs:
```diff
@@ -0,0 +1,174 @@
+using Certify.Service;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Certify.Core.Tests.Unit
```
This is probably an integration test because the defaults will be real data store, real accounts etc. If we were injecting or replacing the respective providers with test ones it might be a unit test. I realize though that there are already plenty of tests in Unit that also fail to make that distinction, but perhaps going forward we should be more pedantic about the distinction between Unit and Integration tests.

jrnelson90 replied:
This has been moved to Integration tests

jrnelson90 committed:
Refactored TestCertifyManagerGetACMEProvider, moved to integration tests

TestCertifyManagerGetACMEProvider now creates/destroys its own ACME account, and grabs the expected Acme Base URI from definition in CertificateAuthority.CoreCertificateAuthorities
